### PR TITLE
chore(*): update dependencies

### PR DESCRIPTION
**Pull request by shkcodes**
Update the dependencies so that the project compiles

### CODE REVIEW

File: buildSrc/src/main/java/Application.kt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 object Application {
     const val applicationId = "com.fueled.technicalchallenge"
-    const val compileSdk = 31
-    const val minSdk = 23
-    const val targetSdk = 31
+    const val compileSdk = 34
+    const val minSdk = 27
+    const val targetSdk = 34
```
Comment (IndianaSloan) [inline buildSrc/src/main/java/Application.kt:5]: I wonder should we keep `targetSdk` as 33 given that 34 is still in beta? But still bump `compileSdk` to satisfy the androidx navigation requirement? (_or happy to leave at 34 seeing as its likely to be released in next few months anyway_)

**Review by IndianaSloan — APPROVED** [review]
Nice work @shkcodes 💫 ✨ Left a small comment, otherwise looks great!

File: buildSrc/src/main/java/Application.kt (hunk 1 above)
Comment (shkcodes) [reply buildSrc/src/main/java/Application.kt:5]: I also had the exact same thought 😄 But considering we are not hiring right now, I assumed by the time any candidate does the exercise, 34 would have released.

Comment (IndianaSloan) [reply buildSrc/src/main/java/Application.kt:5]: yeah thought the same, makes sense!

**Review by libhide — APPROVED** [review]
💯

File: app/build.gradle.kts [hunk 2]
```diff
@@ -25,11 +26,11 @@ android {
...
     compileOptions {
         isCoreLibraryDesugaringEnabled = true
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8.toString()
+        jvmTarget = JavaVersion.VERSION_17.toString()
```
Comment (humblerookie) [inline app/build.gradle.kts]: can we just declare a top level java version variable and reference it here?

File: buildSrc/src/main/java/Application.kt [hunk 3]
```diff
@@ -1,8 +1,8 @@
 object Application {
     const val applicationId = "com.fueled.technicalchallenge"
-    const val compileSdk = 31
-    const val minSdk = 23
-    const val targetSdk = 31
+    const val compileSdk = 34
+    const val minSdk = 27
```
Comment (humblerookie) [inline buildSrc/src/main/java/Application.kt]: 👀 any particular reason to bump the min sdk?

**Review by humblerookie — APPROVED** [review]
Looks good left a few recommendations

File: buildSrc/src/main/java/Application.kt (hunk 3 above)
Comment (shkcodes) [reply buildSrc/src/main/java/Application.kt]: no particular reason, thought I'd bump it as well since this is just an exercise

Comment (humblerookie) [reply buildSrc/src/main/java/Application.kt]: Let's retain the minsdk since it would be better to be consistent with what we use on our projects

Comment (shkcodes) [reply buildSrc/src/main/java/Application.kt]: okay reverting to 26 just like we have in our compose template

**Review by BhavyaRattan — APPROVED** [review]
LGTM!

**Pull request merged**